plugin/phpactor.vim

```diff
@@ -583,31 +583,116 @@ function! phpactor#_rpc_dispatch(actionName, parameters)
 
     " >> file references
     if a:actionName == "file_references"
-        let list = []
+        " if there is only one file, and it is the open file, don't
+        " bother opening the quick fix window
+        if len(a:parameters['file_references']) == 1
+            let fileRefs = a:parameters['file_references'][0]
+            if -1 != match(fileRefs['file'], bufname('%') . '$')
+                return 
+            endif
+        endif
 
+        let results = {}
         for fileReferences in a:parameters['file_references']
             for reference in fileReferences['references']
-                call add(list, {
+                let key = s:relative_path(fileReferences['file'])
+                    \ .':'. reference['line_no']
+                    \ .':'. (reference['col_no'] + 1)
+                    \ .':'. reference['line']
+
+                let results[key] = {
                     \ 'filename': fileReferences['file'],
                     \ 'lnum': reference['line_no'],
                     \ 'col': reference['col_no'] + 1,
                     \ 'text': reference['line']
-                \ })
+                \ }
             endfor
         endfor
 
-        call setqflist(list)
+        if !exists("*fzf#complete") " If fzf.vim is installed, required for with_preview !
+            call <SID>build_quickfix_list(values(results))
 
-        " if there is only one file, and it is the open file, don't
-        " bother opening the quick fix window
-        if len(a:parameters['file_references']) == 1
-            let fileRefs = a:parameters['file_references'][0]
-            if -1 != match(fileRefs['file'], bufname('%') . '$')
-                return 
-            endif
+            return
         endif
 
-        execute ':cwindow'
+        let formated = s:align_pairs(keys(results), '^\(.\{-}:\d\+:\d\+:\)\s*\(.*\)\s*$', 100)
```

Comment:
would it make sense to separate this into a separate function?

Reply:
Possibly even another file/module (however autoloading works) if the new, associated, functions belong only to this strategy?

Reply:
I thought about that.
And I'm kind of relying onto the rule of three for a while now :)
It would make sense but I'm not sure yet in which scenario we could really reuse it.
It's really dependent on the object we provide to the sink, so refactor to make the code more readable yes, to reuse I'm not sure.
Since the all function that handles rcp responses was already like that I didn't even try.
But I could create another strategy like for the lists.
Maybe a"quickfix strategy" of some sort that will require a specific input to be handled properly.

What about opening a new issue for improvement so we can deliver this part already ?

Reply:
I think extracting the method might still be a good idea, not really that concerned about extracting a module `phpactor#fzf#build_quick_fix_list` or whatever is best?

Reply:
Done, I let your resolve if that corresponds to what you were expected